### PR TITLE
feat: add pipeline commands

### PR DESCRIPTION
**Pull request by jbrockopp**
* Added a `PipelineGenerate` command to `go-vela/cli/action`
* Added a `pipelineGenerate()` function called by `PipelineGenerate` command
* Added tests for the `pipelineGenerate()` function
* Added a `PipelineValidate` command to `go-vela/cli/action`
* Added a `pipelineValidate()` function called by `PipelineValidate` command
* Added tests for the `pipelineValidate()` function

### CODE REVIEW

File: action/pipeline_generate.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import (
+	"fmt"
+
+	"github.com/go-vela/cli/action/pipeline"
+
+	"github.com/urfave/cli/v2"
+)
+
+// PipelineGenerate defines the command for producing a pipeline.
+var PipelineGenerate = &cli.Command{
```
Comment (jbrockopp) [inline action/pipeline_generate.go:16]: This is where we add the new `PipelineGenerate` command

File: action/pipeline_generate.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+DOCUMENTATION:
+
+  https://go-vela.github.io/docs/cli/pipeline/generate/
+`, cli.CommandHelpTemplate),
+}
+
+// helper function to capture the provided
+// input and create the object used to
+// produce a pipeline.
+func pipelineGenerate(c *cli.Context) error {
```
Comment (jbrockopp) [inline action/pipeline_generate.go:77]: This is where we add the new `pipelineGenerate()` function

File: action/pipeline_generate_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+package action
+
+import (
+	"flag"
+	"testing"
+
+	"github.com/urfave/cli/v2"
+)
+
+func TestAction_PipelineGenerate(t *testing.T) {
```
Comment (jbrockopp) [inline action/pipeline_generate_test.go:14]: Here are the tests for the `pipelineGenerate()` function

File: action/pipeline_validate.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+	"github.com/go-vela/cli/action/pipeline"
+
+	"github.com/go-vela/compiler/compiler/native"
+
+	"github.com/urfave/cli/v2"
+)
+
+// PipelineValidate defines the command for verifying a pipeline.
+var PipelineValidate = &cli.Command{
```
Comment (jbrockopp) [inline action/pipeline_validate.go:18]: This is where we add the new `PipelineValidate` command

File: action/pipeline_validate.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+DOCUMENTATION:
+
+  https://go-vela.github.io/docs/cli/pipeline/validate/
+`, cli.CommandHelpTemplate),
+}
+
+// helper function to capture the provided
+// input and create the object used to
+// verify a pipeline.
+func pipelineValidate(c *cli.Context) error {
```
Comment (jbrockopp) [inline action/pipeline_validate.go:59]: This is where we add the new `pipelineValidate()` function

File: action/pipeline_validate_test.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+package action
+
+import (
+	"flag"
+	"testing"
+
+	"github.com/urfave/cli/v2"
+)
+
+func TestAction_PipelineValidate(t *testing.T) {
```
Comment (jbrockopp) [inline action/pipeline_validate_test.go:14]: Here are the tests for the `pipelineValidate()` function

**Comment by davidvader** [comment]
reviewed, looks great. 
Just wondering though, is this intended for new Vela users? I think it would be a great feature to promote in our demos

**Pull request merged**